lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,220 @@ def load_team_membership_csv(course, response):
         "Team membership CSV download is not yet implemented."
     )
     response.write(not_implemented_message + "\n")
+
+
+class TeamMemberShipImportManager(object):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self):
+        # the list of validation errors
+        self.error_list = []
+        self.teamset_names_list = []
+        # this is a dictionary of dictionaries that ensures that a student can belong to
+        # one and only one team in a teamset
+        self.teamset_membership_dictionary = {}
```

Comment:
This appears to actually map a teamset's index in the header to a list of user ids.  Also, since you're using this to check existence later, it would be better if the values were sets and not lists.  Lastly, checkout https://docs.python.org/3.5/library/collections.html#collections.defaultdict - you could make this something like `self.user_ids_by_teamset_name = defaultdict(set)`, which would be more efficient and the variable name would make clear exactly what data is stored.